api/tests/routes/backoffice/venues_test.py

```diff
@@ -1406,6 +1409,51 @@ def test_batch_edit_venues_only_criteria(self, legit_user, authenticated_client,
 
         assert set(venues[0].criteria) == {criteria[0], new_criterion}
 
+    @pytest.mark.parametrize(
+        "set_permanent,thumb_count,expected_mail_number",
+        [
+            (True, 0, 1),
+            (False, 0, 0),
+            (True, 1, 0),
+        ],
+    )
+    @patch("pcapi.core.search.async_index_venue_ids")
+    def test_batch_edit_venues_set_permanent_sends_mail(
+        self,
+        mock_async_index_venue_ids,
+        legit_user,
+        authenticated_client,
+        criteria,
+        set_permanent,
+        thumb_count,
+        expected_mail_number,
+    ):
+        venue = offerers_factories.VenueFactory(isPermanent=not set_permanent)
+        venue.thumbCount = thumb_count
+
+        form_data = {
+            "object_ids": str(venue.id),
+            "all_permanent": "on" if set_permanent else "",
+            "all_not_permanent": "" if set_permanent else "on",
+        }
+
+        response = self.post_to_endpoint(authenticated_client, form=form_data)
+        assert response.status_code == 303
+
+        assert venue.isPermanent is set_permanent
+
+        assert len(mails_testing.outbox) == expected_mail_number
+        if expected_mail_number > 0:
+            # check that email is sent when venue is set to permanent has no image
```

Comment:
Erreur de syntaxe, j'ai dû relire plusieurs fois pour comprendre qu'il manque peut-être « and » :)